### PR TITLE
Restructure of data directory

### PR DESCRIPTION
**Pull request by seriva**
- Moved unit tests.
- Moved spec tests.
- Moved ansible sources.
- Move terraform templates.
- Rename data to schema.
- Restructure container builds.
- Fix paths to project wide.
- Fixed Prometheus validation, needed one more tab for a section.
- Added missing tests from @erzetpe from this pull #2576 as her forgot to merge that branch into develop

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,67 +1,66 @@
...
     && rm ./helm-v${HELM_VERSION}-linux-amd64.tar.gz \
     && helm version \
-    && echo "Installing kubectl binary ..." \
+    && : INSTALL KUBECTL BINARY \
     && curl -fsSLO https://storage.googleapis.com/kubernetes-release/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl \
     && chmod +x ./kubectl \
     && mv ./kubectl /usr/local/bin/kubectl \
     && kubectl version --client \
-    && echo "Installing istioctl binary ..." \
+    && : INSTALL ISTIOCTl BINARY \
```
Comment (to-bar) [inline Dockerfile]:
```suggestion
    && : INSTALL ISTIOCTL BINARY \
```

**Review by to-bar — CHANGES_REQUESTED** [review]

**Review by to-bar — DISMISSED** [review]

**Review by to-bar — DISMISSED** [review]

**Comment by przemyslavic** [comment]
This time there is a permission issue:
```
2021-10-06T21:19:57.5553638Z [command]/usr/local/bin/docker run --rm -v /tmp/build/epi-cli:/shared epiphanyregistry.azurecr.io/epicli:merge.5729 -c epicli -h
2021-10-06T21:19:58.1336689Z /bin/bash: line 1: /bin/epicli: Permission denied
2021-10-06T21:19:58.5187486Z ##[error]/bin/bash: line 1: /bin/epicli: Permission denied
2021-10-06T21:19:58.5227687Z ##[error]The process '/usr/local/bin/docker' failed with exit code 126
```

Even if executed from inside the container:
```
epiuser@71f02b052e28:/shared$ epicli -h
bash: /bin/epicli: Permission denied
```

**Comment by seriva** [comment]
> This time there is a permission issue:
> 
> ```
> 2021-10-06T21:19:57.5553638Z [command]/usr/local/bin/docker run --rm -v /tmp/build/epi-cli:/shared epiphanyregistry.azurecr.io/epicli:merge.5729 -c epicli -h
> 2021-10-06T21:19:58.1336689Z /bin/bash: line 1: /bin/epicli: Permission denied
> 2021-10-06T21:19:58.5187486Z ##[error]/bin/bash: line 1: /bin/epicli: Permission denied
> 2021-10-06T21:19:58.5227687Z ##[error]The process '/usr/local/bin/docker' failed with exit code 126
> ```
> 
> Even if executed from inside the container:
> 
> ```
> epiuser@71f02b052e28:/shared$ epicli -h
> bash: /bin/epicli: Permission denied
> ```

mmm not seeing it here under Windows. Will check WSL2.

**Comment by seriva** [comment]
> > This time there is a permission issue:
> > ```
> > 2021-10-06T21:19:57.5553638Z [command]/usr/local/bin/docker run --rm -v /tmp/build/epi-cli:/shared epiphanyregistry.azurecr.io/epicli:merge.5729 -c epicli -h
> > 2021-10-06T21:19:58.1336689Z /bin/bash: line 1: /bin/epicli: Permission denied
> > 2021-10-06T21:19:58.5187486Z ##[error]/bin/bash: line 1: /bin/epicli: Permission denied
> > 2021-10-06T21:19:58.5227687Z ##[error]The process '/usr/local/bin/docker' failed with exit code 126
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Even if executed from inside the container:
> > ```
> > epiuser@71f02b052e28:/shared$ epicli -h
> > bash: /bin/epicli: Permission denied
> > ```
> 
> mmm not seeing it here under Windows. Will check WSL2.

image

image

So yeah, on WSL2 I get the same. Interesting how there still these differences between docker desktop on Windows and WSL2 as they share the same context....

**Comment by seriva** [comment]
> bash: /bin/epicli: Permission denied

image
 Should be fixed.

**Comment by przemyslavic** [comment]
way better 👍

File: epicli [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+export PYTHONPATH=/epicli
+python3 -m cli.epicli $@
```
Comment (to-bar) [inline epicli]:
```suggestion
python3 -m cli.epicli "$@"
```
https://github.com/koalaman/shellcheck/wiki/SC2068

**Review by to-bar — CHANGES_REQUESTED** [review]

File: epicli (hunk 2 above)
Comment (seriva) [reply epicli]: Fixed.

**Review by to-bar — APPROVED** [review]

**Review by przemyslavic — APPROVED** [review]

**Pull request merged**